src/backend/joanie/core/api/client/__init__.py

```diff
@@ -405,52 +420,19 @@ def create(self, request, *args, **kwargs):
                 status=HTTPStatus.BAD_REQUEST,
             )
 
-        # Else return the fresh new order
-        return Response(serializer.data, status=HTTPStatus.CREATED)
-
-    @action(detail=True, methods=["PATCH"])
-    def submit(self, request, pk=None):  # pylint: disable=no-self-use, invalid-name, unused-argument
-        """
-        Submit a draft order if the conditions are filled
-        """
-        billing_address = request.data.get("billing_address")
-        credit_card_id = request.data.get("credit_card_id")
-        order = self.get_object()
-
-        if order.organization is None:
-            order.organization = self._get_organization_with_least_active_orders(
-                order.product, order.course, order.enrollment
-            )
-            order.save()
-
-        return Response(
-            {"payment_info": order.submit(billing_address, credit_card_id)},
-            status=HTTPStatus.CREATED,
+        serializer.instance.flow.assign(
+            billing_address=request.data.get("billing_address")
         )
 
-    @action(detail=True, methods=["POST"])
-    def abort(self, request, pk=None):  # pylint: disable=no-self-use, invalid-name, unused-argument
-        """Change the state of the order to pending"""
-        payment_id = request.data.get("payment_id")
-
-        order = self.get_object()
-
-        if order.state == enums.ORDER_STATE_VALIDATED:
-            return Response(
-                "Cannot abort a validated order.",
-                status=HTTPStatus.UNPROCESSABLE_ENTITY,
-            )
-
-        order.flow.pending(payment_id)
-
-        return Response(status=HTTPStatus.NO_CONTENT)
+        # Else return the fresh new order
+        return Response(serializer.data, status=HTTPStatus.CREATED)
 
     @action(detail=True, methods=["POST"])
     def cancel(self, request, pk=None):  # pylint: disable=no-self-use, invalid-name, unused-argument
         """Change the state of the order to cancelled"""
         order = self.get_object()
 
-        if order.state == enums.ORDER_STATE_VALIDATED:
+        if order.state == enums.ORDER_STATE_COMPLETED:
             return Response(
                 "Cannot cancel a validated order.",
```

Comment:
Should we change also the string response ? 
```suggestion
                "Cannot cancel a completed order.",
```